src/dfastbe/io.py

```diff
@@ -1645,6 +1693,35 @@ def mask(
 
         return search_lines, max_distance
 
+    @staticmethod
+    def _select_closest_part(
+        multiline: MultiLineString,
+        reference_line: LineString,
+        max_river_width: float,
+    ) -> LineString:
+        """
+        Select the closest part of a MultiLineString to the reference line.
+
+        Args:
+            multiline (LineString): The MultiLineString to process.
+            reference_line (LineString): The reference line to calculate distances.
+            max_river_width (float): Maximum allowable distance.
+
+        Returns:
+            LineString: The closest part of the MultiLineString.
+        """
+        closest_part = multiline.geoms[0]
+        min_distance = max_river_width
+
+        for part in multiline.geoms:
+            simplified_part = part.simplify(1)
+            distance = simplified_part.distance(reference_line)
+            if distance < min_distance:
+                min_distance = distance
+                closest_part = part
+
+        return closest_part
+
     def to_polygons(self) -> List[Polygon]:
         """
         Construct a series of polygons surrounding the bank search lines.
```

Comment:
can you add examples here ?

Reply:
can you also check if this is the best way to create a list the `[None] * self.size`

Reply:
check if there is a better way? if not leave it as is